src/mjolnir/ferry_connections.cc

```diff
@@ -177,6 +177,7 @@ std::pair<uint32_t, bool> ShortestPath(const uint32_t start_node_idx,
         }
 
         // if the first edge was destonly and the previous edge was as well, add no penalty
+        // TODO(nils): this would have to take into account HGV destonly to be accurate
```

Comment:
maybe log this situation out

Reply:
I added an implementation of this but not very happy with it. in fact, I'm pretty sure it's the same (bad) behavior as not acting here at all. it's not easy to log out this particular situation either, since `destonly_hgv` also contains normal `destonly`. I'd vote for leaving things as they are.

Reply:
added code to try and handle it: https://github.com/valhalla/valhalla/pull/4450/commits/5395f139feb03be5589cf050e5368cc5aab7722f

I reverted that commit already.